gateway/scaling/function_scaler.go

```diff
@@ -38,6 +38,17 @@ type FunctionScaleResult struct {
 // the minimum replicas metadata
 func (f *FunctionScaler) Scale(functionName string) FunctionScaleResult {
 	start := time.Now()
+
+	if cachedResponse, hit := f.Cache.Get(functionName); hit &&
```

Comment:
This means that a scale can only occur as often as `CacheExpiry` ?

Reply:
If you find where `Scale` is called from (and when) you should see that it encapsulates the whole workflow including whether Scaling is needed. This is instead of two methods for: `IsScaled()` (i.e. should we scale) and `Scale()` (let's scale) being split out into two separate methods.

Reply:
To put it another way - if a cache is not used, it means we'd be calling the provider's function status method for every single invocation of a function which adds considerable latency and resource consumption.